node/src/database/db_initializer.rs

```diff
@@ -496,11 +518,111 @@ pub fn connection_or_panic(
         })
 }
 
+#[derive(Clone)]
+pub struct DbInitializationConfig {
+    pub special_conn_setup: Vec<fn(&Connection) -> rusqlite::Result<()>>,
+    //only things related to db migrations are underneath
+    pub should_be_suppressed: Suppression,
+    pub external_dataset: Option<ExternalData>,
+}
+
+#[derive(PartialEq, Eq, Debug, Clone, Copy)]
+pub enum Suppression {
+    No,
+    Yes,
+    WithErr,
+}
+
+impl DbInitializationConfig {
+    pub fn add_special_conn_setup(
+        mut self,
+        setter: fn(&Connection) -> rusqlite::Result<()>,
+    ) -> Self {
+        self.special_conn_setup.push(setter);
+        self
+    }
+
+    pub fn panic_on_migration() -> Self {
+        Self {
+            special_conn_setup: vec![],
+            should_be_suppressed: Suppression::No,
+            external_dataset: None,
+        }
+    }
+
+    pub fn create_or_migrate(external_params: ExternalData) -> Self {
+        //is used also if a fresh db is being created
+        Self {
+            special_conn_setup: vec![],
+            should_be_suppressed: Suppression::No,
+            external_dataset: Some(external_params),
+        }
+    }
+
+    pub fn migration_suppressed() -> Self {
+        Self {
+            special_conn_setup: vec![],
+            should_be_suppressed: Suppression::Yes,
+            external_dataset: None,
+        }
+    }
+
+    pub fn migration_suppressed_with_error() -> Self {
+        Self {
+            special_conn_setup: vec![],
+            should_be_suppressed: Suppression::WithErr,
+            external_dataset: None,
+        }
+    }
+
+    #[cfg(test)]
+    pub fn test_default() -> Self {
+        Self {
+            special_conn_setup: vec![],
+            should_be_suppressed: Suppression::Yes,
+            external_dataset: Some(ExternalData {
+                chain: TEST_DEFAULT_CHAIN,
+                neighborhood_mode: NeighborhoodModeLight::Standard,
+                db_password_opt: None,
+            }),
+        }
+    }
+}
+
+impl PartialEq for DbInitializationConfig {
+    fn eq(&self, other: &Self) -> bool {
+        (
+            self.external_dataset == other.external_dataset)
+            && (self.should_be_suppressed == other.should_be_suppressed)
+            //I'm making most of this, fn pointers cannot be compared in Rust by August 2022
+            && ((self.special_conn_setup.is_empty() && other.special_conn_setup.is_empty())
```

Comment:
I think you can just compare lengths here, which would make checking for emptiness unnecessary.